pebbles/static/js/controllers/StatsController.js

```diff
@@ -2,9 +2,277 @@ app.controller('StatsController', ['$q', '$scope', '$http', '$interval', '$uibMo
+            }
+            return undefined;
+        }
+        $scope.isCurrentView = function(view){
+            if ($scope.currentView == view){
+                return true;
+            }
+            return false;
+        }
+
+        $scope.totalItems;
+        $scope.visiblePages = 4;
+        $scope.currentPage = 1;
+        var itemsPerPage = 15;
+        $scope.itemsPerPage = itemsPerPage;
+
+        $scope.userlist;
+        $scope.dates = {};
+
+
+        var exportInstitutions = Restangular.all('export_stats/institutions');
+        var exportActiveUsers = Restangular.all('export_stats/active_users');
+        var exportMonthlyInstances = Restangular.all('export_stats/monthly_instances');
+        var exportQuartals = Restangular.all('export_stats/quartals');
+        var exportQuartalsByOrganization = Restangular.all('export_stats/quartals_by_organization')
+
         var stats = Restangular.oneUrl('stats');
+        var activeUsers = Restangular.oneUrl('export_stats/active_users');
+        var inst = Restangular.oneUrl('export_stats/institutions');
+        var monthlyInstances = Restangular.oneUrl('export_stats/monthly_instances');
+        var quartals = Restangular.oneUrl('export_stats/quartals');
+        var quartalsByOrganization = Restangular.oneUrl('export_stats/quartals_by_organization');
+
 
         stats.get().then(function (response) {
             $scope.stats = response;
+        });    
+
+        activeUsers.get($scope.dates).then(function(response){  
+            $scope.activeUsers = response;
+            $scope.totalItems = $scope.activeUsers[0].data.length;
+            $scope.pagination($scope.currentPage);
+        });
+
+        inst.get($scope.dates).then(function (response) {
+            $scope.inst = response;
+        });
+
+        monthlyInstances.get($scope.dates).then(function(response) {
+            $scope.monthlyInstances = response;
+        });
+
+        quartals.get($scope.dates).then(function(response) {
+            $scope.quartals = response;
+        });
+
+        quartalsByOrganization.get($scope.dates).then(function(response) {
+            $scope.quartalsByOrganization = response;   
+                 
+        });
+
+        $scope.pagination = function(page) {
+            var data = $scope.activeUsers[0].data.slice(
+                (page-1) * $scope.itemsPerPage,
+                page * $scope.itemsPerPage
+            );
+            $scope.userlist = data;
+        };
+
+        $scope.$watch('currentPage', function() {
+            if (angular.isDefined($scope.activeUsers)) {
+                $scope.pagination($scope.currentPage);
+            };
+            
         });
-    }]);
+
+        $scope.downloadCSV = function(CSV, fileName) {
+            var blob = new Blob([CSV], {type: 'text/csv'});
+            var anchorLink = document.createElement('a');
+            var mouseEvent = new MouseEvent('click');
+
+            anchorLink.download = fileName + ".csv";
```

Comment:
Doesnt work for me with Internet Explorer 11. Last I checked we should support IE too. (will do code review for this functionaliity after this).

Reply:
Another thing: If the end-date is upto third quarter, the fourth quartar column is also exported with values as 0(even though there are actual values there - misleading)